### PR TITLE
Refactoring of the server skeleton and adding new functionality with the work of a worker downloading

### PR DESCRIPTION
**Pull request by VolodymyrFihurniak**

### CODE REVIEW

File: server/src/business/fileWorker.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
+import { Config } from '@src/entities/config';
 import { Response } from '@src/entities/response';
 import { iFileDatabase } from '@src/interfaces/iFileDatabase';
 import { dataLogger } from '@src/utils/server/logger';
+import { parseURLYouTube } from '@src/utils/server/parseURLYouTube';
```
Comment (denbykov) [inline server/src/business/fileWorker.ts:5]: In name you have to make an accent on the subject of action. For example, here it should be parseYoutubeURL, that way I can understand that you are parsing URL not youtube...

File: server/src/controllers/file.controller.ts [hunk 2]
```diff
@@ -73,6 +74,22 @@ class FileController extends BaseController {
...
     return res.status(picture.httpCode).send(picture.serialize());
   };
+
+  public postURrlFile = async (req: Express.Request, res: Express.Response) => {
+    const fileWorker = new FileWorker(
+      await new FileRepository(this.databasePool),
+      this.config
+    );
+    const token = req.headers.authorization?.split(' ')[1];
+    const decoded = jwt.verify(
```
Comment (denbykov) [inline server/src/controllers/file.controller.ts:84]: Decoded is a bad name for an argument, even if it is clear that it is a token from the context. Rename it to the decodedToken or something like this.

File: server/src/utils/server/publisherAMQP.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            throw errorChannel;
+          }
+          const msg = JSON.stringify(payload);
+          channel.assertQueue(queue, {
+            durable: false,
+          });
+          channel.sendToQueue(queue, Buffer.from(msg), {
+            persistent: true,
+          });
+          dataLogger.info(`[RabbitMQ] Sent ${msg}`);
```
Comment (denbykov) [inline server/src/utils/server/publisherAMQP.ts:35]: Definitely should not have an info severity

Comment (VolodymyrFihurniak) [reply server/src/utils/server/publisherAMQP.ts:35]: Replaced the log level with debug

File: server/src/controllers/file.controller.ts (hunk 2 above)
Comment (VolodymyrFihurniak) [reply server/src/controllers/file.controller.ts:84]: Corrected the name of the variable to decodedToken

File: server/src/business/fileWorker.ts (hunk 1 above)
Comment (VolodymyrFihurniak) [reply server/src/business/fileWorker.ts:5]: Corrected the name of the variable to parseYoutubeURL

**Review by denbykov — APPROVED** [review]

**Pull request merged**